### PR TITLE
Periodic Digest Email frequency should be in days

### PR DESCRIPTION
**Pull request by djuber**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

We're sometimes sending digest emails to users without respecting their frequency preferences.

At first I thought the was tied to not sorting the 10 messages we sampled to compare (it might be, but I doubt it), and then I tried testing it and found that `should_send_email?` was true when the only message was sent even just a few seconds earlier. The setting for periodic email frequency is an integer, and we were comparing it to a difference between two timestamps (in seconds) - so anything farther apart than a default 2 seconds qualified for resending articles.

~Cast the setting to days prior to comparing `(now - last_time) > setting` to ensure we're not doing this more than we intended to.~

Compare the last time an email was sent to a timestamp a few days ago (based on the setting) to determine whether it's time to send the digest email. 

## Related Tickets & Documents

fixes https://github.com/forem/forem/issues/10940

https://github.com/forem/forem/pull/10107 (where I think the issue was introduced)

## QA Instructions, Screenshots, Recordings

```ruby
user = User.first
EmailMessage.create(user: user, mailer: "DigestMailer#digest_email", sent_at: 2.hours.ago)
collector = EmailDigestArticleCollector.new(user)
collector.send(:should_receive_email?)
```

This is expected to be false (2 hours is less than the default 2 days).

```ruby
collector.articles_to_send()
```

This is expected to be empty.

### UI accessibility concerns?

None

## Added/updated tests?

- [x] No, and this is why: I probably should.
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [x] I will share this change internally with the appropriate teams

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Review by jeremyf — APPROVED** [review]

File: app/services/email_digest_article_collector.rb [hunk 1]
```diff
@@ -51,11 +51,14 @@ def should_receive_email?
     return true unless last_email_sent_at
 
     # Has it been at least x days since @user received an email?
-    Time.current - last_email_sent_at >= Settings::General.periodic_email_digest
+    last_email_sent_at < Settings::General.periodic_email_digest.days.ago
```
Comment (citizen428) [inline app/services/email_digest_article_collector.rb:54]: In scenarios like this I really prefer the `before?` and `after?` methods Rails adds:

```suggestion
    last_email_sent_at.before?(Settings::General.periodic_email_digest.days.ago)
```

**Review by citizen428 — APPROVED** [review]

File: app/services/email_digest_article_collector.rb (hunk 1 above)
Comment (jeremyf) [reply app/services/email_digest_article_collector.rb]: OMG, yes!  The mental gymnastics of `>` or `<` with dates is just a bit too much.

tl;dr - TIL about `.before?` and `.after?`

Comment (djuber) [reply app/services/email_digest_article_collector.rb]: this was 100% what I wanted - thanks for the tip
